addons/sys_intercom/fnc_enterVehicle.sqf

```diff
@@ -0,0 +1,67 @@
+/*
+ * Author: ACRE2Team
+ * Handles entering a vehicle
+ *
+ * Arguments:
+ * 0: Vehicle <OBJECT>
+ * 1: Unit entering a vehicle <OBJECT>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [cursorTarget, player] call acre_sys_intercom_fnc_enterVehicle
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+params ["_vehicle", "_unit"];
+
+if (_unit != _vehicle) then {
+    _unit setVariable [QGVAR(intercomVehicle), _vehicle];
+
+    // Configure the array for handling if a player is in a limited intercom position or not.
+    private _usingLimitedPosition = [];
+    {
+        _usingLimitedPosition pushBack false;
+    } forEach (_vehicle getVariable [QGVAR(intercomNames), []]);
+    _unit setVariable [QGVAR(usingLimitedPosition), _usingLimitedPosition];
+
+    {
+        [_vehicle, _unit, _forEachIndex] call FUNC(seatSwitched);
+    } forEach (_vehicle getVariable [QGVAR(intercomNames), []]);
+    GVAR(crewPFH) = [DFUNC(intercomPFH), 1.1, [_unit, _vehicle]] call CBA_fnc_addPerFrameHandler;
+} else {
+    [GVAR(crewPFH)] call CBA_fnc_removePerFrameHandler;
+    private _intercomVehicle = _unit getVariable [QGVAR(intercomVehicle), objNull];
+    private _unitsIntercom = _intercomVehicle getVariable [QGVAR(unitsIntercom) , []];
+    private _disconnected = false;
+    {
+        private _temp = +_x;
```

Comment:
`_temp` is hard for the next person to read the code to understand`. If I understand correctly `_unitsIntercom` represents an array intercoms where each child is an array of units on that intercom. So perhaps `_temp` could be `_unitsOnIntercomX`